api/src/main/java/org/apache/iceberg/ManifestFile.java

```diff
@@ -199,6 +200,12 @@ default boolean hasDeletedFiles() {
      */
     boolean containsNull();
 
+    /**
+     * Returns true if at least one data file in the manifest has a nan value for the field.
+     * Null if this information doesn't exist.
+     */
+    Boolean containsNaN();
```

Comment:
Does it make sense for this to have a default of null?

Reply:
I think once this change is released, this field will be populated by default, and making it nullable is mostly for backward compatibility; also the unassigned value for this field is null in both the [actual implementation](https://github.com/apache/iceberg/pull/1872/files#diff-5ed34e844eaf59fe89785c86989ba8a467414fa112e5d266a26cd867b795bc8bR45) and [test implementation](https://github.com/apache/iceberg/pull/1872/files#diff-c9732959939317302e8738e19b1d831cef747b77d4c8bfa2fdab883fd087a68aR153), so I think defaulting to null here might not help much?

Reply:
While we don't expect other implementations of this interface, it is a fair point that we would accept other implementations in any method that uses `ManifestFile` as an argument type. Adding the default would ensure backward-compatiblity in case anyone has an alternative implementation, and is really low cost. So I'd opt to add it. Thanks @holdenk!